src/screens/DiscoveryScreenV2/hooks/useDiscovery.js

```diff
@@ -138,14 +138,21 @@ const useDiscovery = () => {
   };
 
   const getIsMeFollowingTargetStatus = React.useCallback(
-    (userId) => {
-      const targetUser = discovery?.initialUsers?.find((item) => item?.user_id === userId);
-      const targetUserFromFollowed = discovery?.followedUsers?.find(
-        (item) => item?.user_id === userId
-      );
-      const targetUserFromFollowing = discovery?.unfollowedUsers?.find(
-        (item) => item?.user_id === userId
-      );
+    (userId, username) => {
+      function findUser(user) {
+        const isUsernameSame =
+          user?.username?.toLocaleLowerCase() === username?.toLocaleLowerCase();
+        const isUserIdSame = user?.user_id === userId;
+
+        if (username && username.length > 0) {
+          return isUserIdSame || isUsernameSame;
+        }
+
+        return isUserIdSame;
+      }
+      const targetUser = discovery?.initialUsers?.find(findUser);
+      const targetUserFromFollowed = discovery?.followedUsers?.find(findUser);
+      const targetUserFromFollowing = discovery?.unfollowedUsers?.find(findUser);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made to the `getIsMeFollowingTargetStatus` function are good for improving readability and maintainability. The new helper function `findUser` checks for matching `userId` or `username` in the user objects, which is a more efficient way of finding users. However, there's a potential issue with the current implementation. If the `username` parameter is not provided or is an empty string, the function will only check for matching `userId`. This could lead to incorrect results if there are users with the same `userId` but different usernames. To fix this, you should always check both `userId` and `username`, regardless of whether `username` is provided or not.

```diff
        if (username && username.length > 0) {
          return isUserIdSame || isUsernameSame;
        }

        return isUserIdSame;
```
Change to:
```diff
+        return isUserIdSame || isUsernameSame;
```
This change ensures that the function always checks both `userId` and `username`, providing more accurate results.


<!-- This is an auto-generated comment by OSS CodeRabbit -->